src/pricecypher/contracts/ScopeScript.py

```diff
@@ -0,0 +1,35 @@
+from abc import ABC, abstractmethod
+from typing import Any, Optional
+
+from pricecypher.contracts.Script import Script
+
+
+class ScopeScript(Script, ABC):
+    """
+    The abstract ScopeScript class serves as an interaction contract such that by extending it with its
+        methods implemented, a script can be created that calculates values for some scope for transactions,
+        which can then be used in a generalized yet controlled setting.
+    """
+
+    def execute(self, business_cell_id: Optional[int], bearer_token: str, user_input: dict[Any: Any]) -> Any:
+        # Executing a scope-script like a normal script:
+        # Attempt to extract the transaction IDs and continue as normal
+        transaction_ids: list[int] = user_input['transaction_ids']
```

Comment:
I'd suggest adding a check before this to make sure that the user input does indeed contain the transaction IDs key, and give a meaningful error message otherwise

Reply:
Good idea, added